olcut-core/src/main/java/com/oracle/labs/mlrg/olcut/command/CommandInterpreter.java

```diff
@@ -1508,6 +1509,74 @@ public String unredir(CommandInterpreter ci) {
+         * constraint.
+         *
+         * @param ci the command interpreter
+         * @param level the new level to set the log to
+         * @param loggerName the name of a logger (expects package or canonical class name)
+         * @return a string of command output
+         */
+        @Command(usage="<LogLevel> [<LoggerName>] - sets the logging level for a logger, if specified, or sets the root handler level")
+        public String setLogLevel(CommandInterpreter ci, String level, @Optional(val="") String loggerName) {
+            Level logLevel = null;
+            try {
+                //
+                // Verify our args.
+                logLevel = Level.parse(level);
+
+                //
+                // Logger names by convention are class or package names. Warn in that isn't the case to help
+                // prevent typo frustration.
+                if (!loggerName.isEmpty()) {
+                    // Is this a class name?
+                    try {
+                        Class.forName(loggerName);
+                    } catch (ClassNotFoundException e) {
+                        // Not a class - how about package name?
+                        if (!Arrays.stream(Package.getPackages())
+                                .anyMatch(p -> p.getName().startsWith(loggerName) && (p.getName().equals(loggerName) || p.getName().charAt(loggerName.length()) == '.'))) {
+                            // Not a package either
+                            ci.out.println("Warning: \"" + loggerName + "\" doesn't appear to be a class or package name");
+                        }
+                    }
+                    Logger target = Logger.getLogger(loggerName);
+                    target.setLevel(logLevel);
+                }
+
+                //
+                // Get the current Handler level.
+                Handler[] handlers = Logger.getLogger("").getHandlers();
+                if (handlers == null || handlers.length <= 0) {
+                    return "No root log handlers defined.";
+                }
+                Level currLevel = handlers[0].getLevel();
+
+                //
+                // Should we adjust the handler level too?  We should if no logger was specified
+                // or if the requested level is finer than the current level.
+                if (loggerName.isEmpty() || (logLevel.intValue() < currLevel.intValue())) {
+                    for (Handler h : handlers) {
+                        h.setLevel(logLevel);
+                    }
+                } else if (!loggerName.isEmpty() && logLevel.intValue() > currLevel.intValue()) {
+                    ci.out.println(logLevel.toString() + " is less fine than current "
+                            + currLevel.toString() + ", won't automatically deescalate the Handler level");
+                }
+            } catch (IllegalArgumentException e) {
```

Comment:
Do we want the try to be this broad? Are there other things in here that could throw `IllegalArgumentException`?

Reply:
I don't think anything else in there will throw that exception, but fair point anyway - I narrowed the scope of the try catch just to surround that parse.